pulp_container/app/tasks/sync_stages.py

```diff
@@ -648,4 +649,4 @@ def _post_save(self, batch):
         # it is possible to initialize the nature of the corresponding manifest lists
         for ml in manifest_lists:
             if ml.init_manifest_list_nature():
-                ml.save(update_fields=["is_bootable", "is_flatpak"])
+                ml.save(update_fields=["is_bootable", "is_flatpak", "nature"])
```

Comment:
_post_save runs in the same transaction as the things are created, right?

I still wonder, can we not know all this before we even save the manifests the first time?

Reply:
I updated the current PR to address this concern, but I didn't modify the current implementation `ml.save(update_fields=["is_bootable", "is_flatpak"])` to avoid introducing bugs (and this block will be removed "soon" since these fields are now deprecated).